### PR TITLE
ci: Add dependabot go-storage auto build support

### PR DESCRIPTION
**Pull request by JinnyYi**

### CODE REVIEW

File: .github/workflows/go-storage-auto-build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.1.0
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Check dependency names
```
Comment (Xuanwo) [inline .github/workflows/go-storage-auto-build.yml:19]: How about removing this section directly?

Comment (Xuanwo) [reply .github/workflows/go-storage-auto-build.yml:19]: It's looks not harmful for other deps to build again.

Comment (JinnyYi) [reply .github/workflows/go-storage-auto-build.yml:19]: OK. Does the file need to be renamed to `auto-build.yml`?

Comment (Xuanwo) [reply .github/workflows/go-storage-auto-build.yml:19]: > OK. Does the file need to be renamed to `auto-build.yml`?

LGTM

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**